tests/tensor/test_nlinalg.py

```diff
@@ -120,6 +123,117 @@ def test_matrix_dot():
     assert _allclose(numpy_sol, aesara_sol)
 
 
+class TestMatMul(utt.InferShapeTester):
+    def setup_method(self):
+        super().setup_method()
+        self.rng = np.random.default_rng(utt.fetch_seed())
+        self.op = matmul
+        self.op_class = MatMul
```

Comment:
This use of a shared RNG state induces test method order dependence; instead, if you create and seed the RNG objects within each independent test unit, they can be run in any order (and in parallel) and produce consistent results.

Reply:
Just to be clear: by "test unit" you mean each independent `test_*` method?

Reply:
> Just to be clear: by "test unit" you mean each independent `test_*` method?

Yes

Reply:
It looks like all the tests are sharing the same class-level RNG object.  As I mentioned earlier, this will make the results order-dependent.  We need to construct the RNG object within each individual test in order to avoid that.

N.B. A fixture could be used if you don't want to copy-paste the RNG construction code each time.